drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -20,8 +21,68 @@
 
 LOG_MODULE_REGISTER(siwx91x_wifi);
 
+#define SIWX91X_INTERFACE_MASK (0x03)
+
 NET_BUF_POOL_FIXED_DEFINE(siwx91x_tx_pool, 1, _NET_ETH_MAX_FRAME_SIZE, 0, NULL);
 
+static inline int siwx91x_convert_mode(sl_wifi_interface_t sl_iface)
+{
+	switch (sl_iface) {
+	case SL_WIFI_CLIENT_2_4GHZ_INTERFACE:
+		return WIFI_STA_MODE;
+	case SL_WIFI_AP_2_4GHZ_INTERFACE:
```

Comment:
Please avoid symbols `SL_WIFI_(AP|CLIENT)_2_4GHZ_INTERFACE`. They are misleading. Here, the relevant data is probably: `FIELD_GET(SIWX91X_INTERFACE_MASK, sl_iface)`